### PR TITLE
feat(pci-workflow): optimize tests and use pci-common

### PR DESCRIPTION
**Pull request by kqesar**
ref: DTCORE-2614

| Question         | Answer
| ---------------- | ---
| Branch?          | `feat/pci-workflow`
| Bug fix?         | no
| New feature?     | no
| Breaking change? | no
| Tickets          | DTCORE-2614
| License          | BSD 3-Clause

- [ ] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [ ] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [ ] Standalone app was ran and tested locally
- [ ] Ticket reference is mentioned in linked commits (internal only)
- [ ] Breaking change is mentioned in relevant commits

### CODE REVIEW

File: packages/manager/apps/pci-kubernetes/src/App.tsx [hunk 1]
```diff
@@ -1,4 +1,8 @@
-import { RouterProvider, createHashRouter } from 'react-router-dom';
+import {
+  RouterProvider,
+  createHashRouter,
```
Comment (frenautvh) [inline packages/manager/apps/pci-kubernetes/src/App.tsx:3]:
```suggestion
```

File: packages/manager/modules/manager-pci-common/src/api/data/instance.ts [hunk 2]
```diff
@@ -23,6 +23,9 @@ export type TInstance = {
   region: string;
   sshKeyId?: string;
   status: string;
+  statusGroup?: string;
+  regionName?: string;
+  flavorName?: string;
```
Comment (frenautvh) [inline packages/manager/modules/manager-pci-common/src/api/data/instance.ts:28]: if you check the api schema those properties does not exists :
https://eu.api.ovh.com/console/?section=%2Fcloud&branch=v1#get-/cloud/project/-serviceName-/instance

you have to add those props in the pci-workflow app using a new type but the types in pci-common should reflect the api schema

File: packages/manager/modules/manager-pci-common/src/api/hook/useIntance.ts
Comment (anooparveti) [inline packages/manager/modules/manager-pci-common/src/api/hook/useIntance.ts:1]: Correct the file name. There is a typo `useInstance`.

**Review by anooparveti — CHANGES_REQUESTED** [review]

File: packages/manager/modules/manager-pci-common/src/api/data/instance.ts [hunk 3]
```diff
@@ -1,6 +1,6 @@
 import { v6 } from '@ovh-ux/manager-core-api';
 
-export type TInstance = {
+export type TInstanceAPI = {
```
Comment (frenautvh) [inline packages/manager/modules/manager-pci-common/src/api/data/instance.ts:3]: i think we should keep TInstance type as API is implicit,
in your app you shoud TWorkflowInstance for example if you need to add custom fields

File: packages/manager/apps/pci-workflow/src/components/DeleteWorkflowRedirect.spec.tsx [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    expect(container.innerHTML).toContain(
+      '/pci/projects/mocked_projectId/workflow/delete/workflowId',
+    );
+  });
+
+  it('handles missing projectId gracefully', () => {
+    vi.mocked(useParams).mockReturnValue({});
+
+    const { container } = render(
+      <MemoryRouter initialEntries={['/initial']}>
```
Comment (seven-amid) [inline packages/manager/apps/pci-workflow/src/components/DeleteWorkflowRedirect.spec.tsx:31]: Why MemoryRouter, and not HashRouter used by the App ?

File: packages/manager/modules/manager-pci-common/src/api/hook/useInstance.ts [hunk 5]
```diff
@@ -0,0 +1,19 @@
+import { useQuery } from '@tanstack/react-query';
+import { getInstance, getInstances } from '../data/instance';
+
+export const useInstance = (projectId: string, instanceId: string) =>
+  useQuery({
+    queryKey: ['instances', instanceId],
```
Comment (seven-amid) [inline packages/manager/modules/manager-pci-common/src/api/hook/useInstance.ts:6]: maybe add the projectId to the queryKey.

File: packages/manager/apps/pci-workflow/src/components/DeleteWorkflowRedirect.spec.tsx [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+describe('DeleteWorkflowRedirect', () => {
+  it('redirects to the correct URL with projectId and workflowId', () => {
+    const { container, debug } = render(
+      <MemoryRouter initialEntries={['/initial']}>
+        <Routes>
+          <Route path="/initial" element={<DeleteWorkflowRedirect />} />
+        </Routes>
+      </MemoryRouter>,
+    );
+    debug(container);
```
[the remark this answers is not in the document]
Comment (kqesar) [reply packages/manager/apps/pci-workflow/src/components/DeleteWorkflowRedirect.spec.tsx]: I am missed to delete it, thank you !

**Review by sidlynx — APPROVED** [review]

**Pull request merged**